app/bundles/EmailBundle/Swiftmailer/Transport/CallbackTransportInterface.php

```diff
@@ -0,0 +1,35 @@
+<?php
+
+/*
+ * @copyright   2015 Mautic Contributors. All rights reserved
+ * @author      Mautic
+ *
+ * @link        http://mautic.org
+ *
+ * @license     GNU/GPLv3 http://www.gnu.org/licenses/gpl-3.0.html
+ */
+
+namespace Mautic\EmailBundle\Swiftmailer\Transport;
+
+use Mautic\CoreBundle\Factory\MauticFactory;
+use Symfony\Component\HttpFoundation\Request;
+
+interface CallbackTransportInterface
+{
+    /**
+     * Returns a "transport" string to match the URL path /mailer/{transport}/callback.
+     *
+     * @return mixed
+     */
+    public function getCallbackPath();
+
+    /**
+     * Handle response.
+     *
+     * @param Request       $request
+     * @param MauticFactory $factory
+     *
+     * @return array array('bounces' => array('hashID' => 'reason', ...));
```

Comment:
Could it return a data object?

Reply:
It could but it'd be a BC break. All I did here was rename the class to meet symfony standards.